lstchain/datachecks/dl1_checker.py

```diff
@@ -0,0 +1,885 @@
+"""
+Functions to check the contents ofLST DL1 files and associated muon ring files
+"""
+
+__all__ = [
+    'check_dl1',
+    'process_dl1_file',
+    'plot_datacheck',
+    'plot_trigger_types',
+    'plot_mean_and_stddev',
+]
+
+import h5py
+import matplotlib.colors as colors
+import matplotlib.dates as dates
+import matplotlib.pyplot as plt
+import matplotlib.ticker as mtick
+import numpy as np
+import os
+import pandas as pd
+import tables
+
+from astropy import units as u
+from astropy.table import Table, vstack
+from ctapipe.coordinates import EngineeringCameraFrame
+from ctapipe.instrument import CameraGeometry
+from ctapipe.io import HDF5TableWriter
+from ctapipe.visualization import CameraDisplay
+from datetime import datetime
+from lstchain.datachecks.containers import DL1DataCheckContainer
+from lstchain.datachecks.containers import DL1DataCheckHistogramBins
+from lstchain.io.io import dl1_params_lstcam_key
+from matplotlib.backends.backend_pdf import PdfPages
+from multiprocessing import Pool
+from scipy.stats import poisson, sem
+from sys import platform
+
+
+def check_dl1(filenames, output_path, max_cores=4):
+    """
+
+    Parameters
+    ----------
+    filenames: _sorted_ (by growing subrun index) list of input DL1 .h5 files
+    output_path: directory where output will be written
+    max_cores: maximum number of processes that the function will spawn (each
+    processing a different subrun)
+
+    Returns
+    -------
+    None
+
+    """
+
+    # to allow the use of multi-processing in linux:
+    if platform == 'linux':
+        os.system("taskset -p 0xff %d" % os.getpid())
+
+    # obtain run number, and first part of file name, from first file:
```

Comment:
This is the nth different version of parsing that file name pattern. I'll open a PR with better way of doing this.